### PR TITLE
grpcjava

### PR DESCRIPTION
**Pull request by ericwenn**
- chore(grpcjava): bump to v1.45.1
- fix(grpcjava): remove invalid check for version

### CODE REVIEW

**Review by thall — APPROVED** [review]
LGTM :+1:

File: tools/sggrpcjava/tools.go [hunk 1]
```diff
@@ -27,15 +25,6 @@ func PrepareCommand(ctx context.Context) error {
...
 	binDir := sg.FromToolsDir("grpc-java", version, "bin")
 	binary := filepath.Join(binDir, binaryName)
-	// read the whole pom at once
-	b, err := os.ReadFile("pom.xml")
-	if err != nil {
-		panic(err)
-	}
-	s := string(b)
-	if !strings.Contains(s, fmt.Sprintf("<grpc.version>%s", version)) {
-		return fmt.Errorf("pom.xml is out of sync with gRPC Java version - expecting %s", version)
```
[the remark this answers is not in the document]
Comment (ericwenn) [reply tools/sggrpcjava/tools.go:37]: Yes, this is similar to what we are using for other protoc dependencies.

[the remark this answers is not in the document]
Comment (ericwenn) [reply tools/sggrpcjava/tools.go:37]: See this for example:
https://github.com/einride/sage/blob/master/tools/sgprotocgengogrpc/tools.go

[the remark this answers is not in the document]
Comment (viktorvoltaire) [reply tools/sggrpcjava/tools.go:37]: :+1:

**Pull request merged**